app/services/chunkers.rb

```diff
@@ -0,0 +1,35 @@
+module Chunkers
+  class UnknownChunkingMethod < StandardError; end
+
+  include Enumerable
+
+  module InputType
+    PLAIN_TEXT = 0
+    COMMON_MARK = 1
+  end
+
+  CHUNKING_METHODS = [
+    { id: :recursive_split, name: "Recursive Split" },
```

Comment:
What do you think of just having the snake case name and using '.titleize' (and eventually translations) to get the view string?

Reply:
I considered that; however I wasn't sure if the user-friendly name was always going to map that well to the symbol. While internally we may use a "Recursive Character Splitter", for a user those words mean nothing and we may instead want to say something else. 

That aside, I chose the "array of records" because I wanted to add some more properties. E.g. defaults for chunk size and overlap. In addition, some other chunking methods don't require chunk size or overlap as parameters, or add others.

Reply:
If we're not going to use the user-friendly name in the UI, what is it for?